server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/SameShardAllocationDecider.java

```diff
@@ -86,19 +85,13 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
             return YES_AUTO_EXPAND_ALL;
         }
         if (node.node() != null) {
-            assert Strings.hasLength(node.node().getHostAddress()) : node;
```

Comment:
I would like to leave the assert in place for a bit to ensure we have plenty CI runs on this.